### PR TITLE
NXP-24967: allow saved searches to be loaded via uid url

### PR DESCRIPTION
**Pull request by Gabez0r**

### CODE REVIEW

File: elements/nuxeo-app.js [hunk 1]
```diff
@@ -666,6 +690,19 @@ Polymer({
     this.$.doc
       .get()
       .then((doc) => {
+        if (this.docId && doc.facets.includes('SavedSearch')) {
+          this._routedSearch = doc;
+          const savedSearchForm = this._getSavedSearchForm();
+          if (savedSearchForm) {
+            const name = savedSearchForm.getAttribute('search-name');
+            this.navigateTo('search', name);
+            if (!this.__searchOnLoad) {
```
Comment (nmpcunha) [inline elements/nuxeo-app.js]: Should this be `!this._searchOnLoad`?

Comment (Gabez0r) [reply elements/nuxeo-app.js]: Yes, because if `this._searchOnLoad` is true, than `_loadSavedSearch` is expected to run on `_updateSearch` (where we set `_searchOnLoad` to `false`).

Comment (Gabez0r) [reply elements/nuxeo-app.js]: ah, sorry, it's the double `_`. Will fix it, thanks!

**Comment by RSalem07** [comment]
I now this is not related to your modification :). Should we define the `searches` property on the `nuxeo-search-form` element ->  
[here](https://github.com/nuxeo/nuxeo-web-ui/blob/b6de55a7dbbc06a207cf87acbcda317aae53b33d/elements/search/nuxeo-search-form.js#L253)

**Comment by Gabez0r** [comment]
@RSalem07 Yes, I agree. For clarity and consistency it should be declared, although we would have to rename it since it would most likely become a private property. Feel free to create a cleanup ticket. 👍

**Comment by Gabez0r** [comment]
As for the second comment about the `this`, I'd say that it's a left over from a previous version of the element (from 10.10). We should double check and remove it. We can re-use the same clean up ticket as described above.

**Comment by RSalem07** [comment]
thanks for taking into account the add of the `search` field and the remove of the bind call :)

File: elements/search/nuxeo-search-form.js [hunk 2]
```diff
@@ -905,4 +911,26 @@ Polymer({
   _displayQuickFilters() {
     return this._quickFilters && this._quickFilters.length > 0;
   },
+
+  _loadSavedSearch(id) {
+    if (id) {
```
Comment (andreacornaglia) [inline elements/search/nuxeo-search-form.js]: can we flatten out this function? It's a little bit hard to follow the logic of ifs/elses.  My suggestions:
if(!id) return;
const load = () => { if(this.searches){ return} ...}
if (this.results) return load();
this._loadSavedSearchListener

Comment (Gabez0r) [reply elements/search/nuxeo-search-form.js]: Thank you for the suggestion. I'll make an early return to reduce indentation:
```
    if (!id) {
      return;
    }
```
But we usually do not use single-line if blocks because they can lead to errors when someone whats to add a a second statement, and they force people to add the curly-braces afterwards, making diffs more messy.

**Comment by Gabez0r** [comment]
Pipeline check: https://qa.nuxeo.org/jenkins/view/All/job/web-ui-pipeline-gbarata/327/ 🔵

File: elements/search/nuxeo-search-form.js (hunk 2 above)
Comment (andreacornaglia) [reply elements/search/nuxeo-search-form.js]: yes! single line was just to save space in the comment :)
